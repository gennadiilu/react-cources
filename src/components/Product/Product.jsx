import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

const displayName = 'Product';

const propTypes = {
  name: PropTypes.string.isRequired,
  handleRemove: PropTypes.func.isRequired
};

function Product({ id, name, handleRemove, isRemoved }) {
  return (
    name && (
      <div data-qa={`product-${id}`}>
        {name}
        -
        <Button text="Remove" onClick={handleRemove} value="removeProduct" />
        Removed - 
        {' '}
        {isRemoved ? 'true' : 'false'}
      </div>
    )
  );
}

Product.displayName = displayName;
Product.propTypes = propTypes;

export default Product;
