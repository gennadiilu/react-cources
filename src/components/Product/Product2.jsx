import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

const displayName = 'Product';

const propTypes = {
  name: PropTypes.string.isRequired,
  handleRemove: PropTypes.func.isRequired
};

function Product2({ id, name, handleRemove }) {
  return (
    name && (
      <div data-qa={`product-${id}`}>
        {name}
        -
        <Button text="Remove" onClick={handleRemove} value="removeProduct" />
      </div>
    )
  );
}

Product2.displayName = displayName;
Product2.propTypes = propTypes;

export default Product2;
