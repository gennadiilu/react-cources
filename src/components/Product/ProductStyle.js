import styled from 'styled-components';

const ProductStyle = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100px;
  height: 100px;
  border: 1px solid ${({ theme }) => theme.borderColor};

  @media {
  }
`;

export default ProductStyle;
