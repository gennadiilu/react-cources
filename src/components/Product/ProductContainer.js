import {
  compose,
  setDisplayName,
  withHandlers,
  //  renderComponent,
  setPropTypes,
  pure,
  //  branch,
  withState,
  withProps
} from 'recompose';
import PropTypes from 'prop-types';
import Product from './Product';
// import Product2 from './Product2';

export const enhance = compose(
  setDisplayName('ProductContainer'),
  withProps({ userType: 'vip' }),
  setPropTypes({
    id: PropTypes.number.isRequired,
    onRemove: PropTypes.func.isRequired
  }),
  withState('isRemoved', 'setIsRemoved', false),
  withHandlers({
    handleRemove: ({ id, onRemove, setIsRemoved }) => () => {
      setIsRemoved(true);
      onRemove(id);
    }
  }),
  pure
  // branch(({ userType }) => userType === 'vip', renderComponent(Product))
);

export default enhance(Product);
