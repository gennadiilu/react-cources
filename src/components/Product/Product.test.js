import React from 'react';
import { shallow } from 'enzyme';
import Product from './Product';

describe('Given the Product component', () => {
  describe('when the component is rendered', () => {
    const component = shallow(<Product id={1} name="Product 1" handleRemove={jest.fn()} />);

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
