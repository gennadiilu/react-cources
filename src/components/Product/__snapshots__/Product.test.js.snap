// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Product component when the component is rendered should match the snapshot 1`] = `
<styled.div
  data-qa="product-1"
>
  <styled.span>
    Product 1
  </styled.span>
  <withRouter(Button)
    onClick={[MockFunction]}
    text="Remove"
    value="removeProduct"
  />
</styled.div>
`;
